kortex_driver/src/hardware_interface.cpp

```diff
@@ -841,76 +818,66 @@ void KortexMultiInterfaceHardware::readGripperPosition()
 return_type KortexMultiInterfaceHardware::write(
   const rclcpp::Time & /*time*/, const rclcpp::Duration & /*period*/)
 {
-  if (block_write)
+  if (in_fault_)
   {
-    feedback_ = base_cyclic_.RefreshFeedback();
-    return return_type::OK;
+    RCLCPP_ERROR(LOGGER, "Error! The hardware is in a fault state, unable to write commands.");
+    return hardware_interface::return_type::ERROR;
   }
 
-  if (!std::isnan(reset_fault_cmd_) && fault_controller_running_)
+  if (block_write)
   {
-    resetFaults();
+    feedback_ = base_cyclic_.RefreshFeedback();
+    return return_type::OK;
   }
 
-  if (in_fault_ == 0.0)
+  // Twist controller active?
+  if ((arm_mode_ == k_api::Base::ServoingMode::SINGLE_LEVEL_SERVOING))
   {
-    if (arm_mode_ == k_api::Base::ServoingMode::SINGLE_LEVEL_SERVOING)
+    if (twist_controller_running_)
```

Comment:
Do we want to change all this logic if we're not going to test the twist controller?